### PR TITLE
refactor!: Use `Delegate` for track linearizers

### PR DESCRIPTION
**Pull request by paulgessinger**
This allows removing the linearizer template from: 
- Billoir fitter
- ZScan finder
- IVF
- AMVFitter

Part of:
- #2842 

Blocked by:
- #2886

### CODE REVIEW

File: Core/include/Acts/Vertexing/TrackLinearizer.hpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#include "Acts/Utilities/Delegate.hpp"
+
+namespace Acts {
+struct LinearizedTrack;
+class Surface;
+
+using TrackLinearizer = Acts::Delegate<Result<LinearizedTrack>(
+    const BoundTrackParameters&, double, const Surface& perigeeSurface,
+    const Acts::GeometryContext&, const Acts::MagneticFieldContext&,
+    MagneticFieldProvider::Cache&)>;
```
Comment (andiwand) [inline Core/include/Acts/Vertexing/TrackLinearizer.hpp:24]:
```suggestion
    const BoundTrackParameters& parameters, double, const Surface& perigeeSurface,
    const Acts::GeometryContext& geoContext, const Acts::MagneticFieldContext& magContext,
    MagneticFieldProvider::Cache& magProvider)>;
```

**Review by andiwand — APPROVED** [review]
Makes it much cleaner to have the linearizer als a config property

File: Core/include/Acts/Vertexing/TrackLinearizer.hpp (hunk 1 above)
Comment (paulgessinger) [reply Core/include/Acts/Vertexing/TrackLinearizer.hpp:24]: I did not know you could do that, actually.

Comment (paulgessinger) [reply Core/include/Acts/Vertexing/TrackLinearizer.hpp:24]: I'll model it after the argument names in `HelicalTrackLinearizer` now.

**Review by andiwand — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [00025676f4d93d91fc3c20905bc2363f3daa287a]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/36188935) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6906310)!

**Pull request merged**